src/netconfig.cpp

```diff
@@ -0,0 +1,374 @@
+
+    puts(completeMessage);
+}
+
+/** @brief Add/remove IP: `ip [VLANID] {add|del} IP[/MASK GATEWAY]` */
+static void cmdIp(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Action action = args.asAction();
+
+    if (action == Action::add)
+    {
+        const auto [ipVer, ip, mask] = args.asIpAddrMask();
+        const auto [gwVer, gwIp] = args.asIpAddress();
+        args.expectEnd();
+        if (ipVer != gwVer)
+        {
+            throw std::invalid_argument("IP version mismatch");
+        }
+
+        const char* ipInterface =
+            ipVer == IpVer::v4 ? Dbus::ip4Interface : Dbus::ip6Interface;
+
+        bus.call(object.c_str(), Dbus::ipCreateInterface, Dbus::ipCreateMethod,
+                 ipInterface, ip, mask, gwIp);
+    }
+    else
+    {
+        const auto [_, ip] = args.asIpAddress();
+        args.expectEnd();
+
+        // Search for IP address' object
+        bool handled = false;
+        for (const auto& it : bus.getAddresses(object.c_str()))
+        {
+            if (it.address == ip)
+            {
+                bus.call(it.object.c_str(), Dbus::deleteInterface,
+                         Dbus::deleteMethod);
+                handled = true;
+                break;
+            }
+        }
+        if (!handled)
+        {
+            std::string err = "IP address ";
+            err += ip;
+            err += " not found";
+            throw std::invalid_argument(err);
+        }
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP client: 'dhcp [VLANID] {enable|disable}` */
+static void cmdDhcp(Dbus& bus, Arguments& args)
+{
+    const std::string object = pathFromOptional(args);
+    const Toggle toggle = args.asToggle();
+    args.expectEnd();
+
+    const bool enable = toggle == Toggle::enable;
+    printf("%s DHCP client...\n", enable ? "Enable" : "Disable");
+
+    bus.set(object.c_str(), Dbus::ethInterface, Dbus::ethDhcpEnabled, enable);
+
+    puts(completeMessage);
+}
+
+/** @brief Enable/disable DHCP features: 'dhcpcfg {enable|disable} {dns|ntp}` */
+static void cmdDhcpcfg(Dbus& bus, Arguments& args)
+{
+    const char* featureDns = "dns";
+    const char* featureNtp = "ntp";
+
+    const Toggle toggle = args.asToggle();
+    const char* feature = args.asOneOf({featureDns, featureNtp});
+    args.expectEnd();
+
+    const bool enable = toggle == Toggle::enable;
+
+    if (strcmp(feature, featureDns) == 0)
+    {
+        printf("%s DNS over DHCP...\n", enable ? "Enable" : "Disable");
+        bus.set(Dbus::objectDhcp, Dbus::dhcpInterface, Dbus::dhcpDnsEnabled,
+                enable);
+    }
+    else
+    {
+        printf("%s NTP over DHCP...\n", enable ? "Enable" : "Disable");
+        bus.set(Dbus::objectDhcp, Dbus::dhcpInterface, Dbus::dhcpNtpEnabled,
+                enable);
+    }
+
+    puts(completeMessage);
+}
+
+/** @brief Add/remove DNS server: `dns [VLANID] {add|del} [static] IP` */
```

Comment:
I don't know how a DNS server may be bound to a VLANID. The kernel decides what interface to use for a traffic to a specific IP address based on the routing tables built from IP/netmask configuration of interfaces.

Reply:
Actually, I have no experience with VLANs, so I can't say how it should work.

Reply:
I do have experience with VLANs. Please remove `VLANID` from this command. DNS is a global setting.